mlir/lib/Dialect/Vector/Transforms/LowerVectorMultiReduction.cpp

```diff
@@ -310,9 +310,16 @@ struct TwoDimMultiReductionToElementWise
                                 PatternRewriter &rewriter) const override {
     auto maskableOp =
         cast<vector::MaskableOpInterface>(multiReductionOp.getOperation());
-    if (maskableOp.isMasked())
-      // TODO: Support masking.
-      return failure();
+
+    Operation *rootOp;
+    Value mask = nullptr;
+    if (maskableOp.isMasked()) {
+      rewriter.setInsertionPoint(maskableOp.getMaskingOp());
```

Comment:
Can we move this lower to when we start creating operations and just add rewriter.setInsertionPoint(rootOp) ?

Reply:
we should also add an insertion guard to restore the prev insertion point.

Reply:
moved it a bit lower but I d like to obtain the mask inside the same conditional.. hence it cant be at where we start creating operations -- consistent with all other masked implementation in this file.

thanks for catching insertion guard. fixed now.